### PR TITLE
NEWS: add release notes for 0.28

### PR DESCRIPTION
**Pull request by garlick**
Candidate release notes - feel free to push changes directly to this branch.

### CODE REVIEW

**Comment by garlick** [comment]
Updated for a release this afternoon.

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,47 @@
...
+ * libflux: make `flux_plugin_handler` topic member const (#3720)
+ * job-manager: add builtin begin-time dependency plugin (#3704)
+ * broker: send offline responses while broker is initializing (#3712)
+ * python: add `flux.util.parse_datetime` (#3711)
+ * job-manager: support simple `after*` job dependencies (#3696)
+ * jobtap: fixes and api enhancements to support dependency plugins (#3698)
+ * shell: add exit-on-error option (#3692)
+
+### Cleanup/Testing/Build System
+ * job-manager: minor cleanup and improvements for event handling #3759
```
Comment (grondo) [inline NEWS.md:29]: very minor nit, PR issue number not enclosed in parenthesis.
```suggestion
 * job-manager: minor cleanup and improvements for event handling (#3759)
```

Comment (garlick) [reply NEWS.md:29]: oops!  I'll fix.

**Review by grondo — APPROVED** [review]
Looks great! Thanks! I'll set MWP

**Comment by grondo** [comment]
Had to restart CI due to a timeout in the `test-install` builder.

**Pull request merged**